### PR TITLE
Feat: integrate CAMEL agent as a backend model

### PR DESCRIPTION
**Pull request by WHALEEYE**
Integrate agents in CAMEL into CRAB as a backend model.

### CODE REVIEW

File: crab/agents/backend_models/camel_model.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+def find_model_type(model_name: str) -> Union[ModelType, str]:
+    for model in ModelType:
+        if model.value.lower() == model_name.lower():
+            return model
+    return model_name
+
+
+def decode_image(encoded_image: str) -> Image:
+    data = base64.b64decode(encoded_image)
+    return Image.open(io.BytesIO(data))
```
Comment (dandansamax) [inline crab/agents/backend_models/camel_model.py:55]: We have a common decoding function in `crab.utils.common`

File: crab/agents/backend_models/camel_model.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            history_messages_len,
+        )
+
+    def get_token_usage(self):
+        return self.token_usage
+
+    def reset(self, system_message: str, action_space: Optional[List[Action]]) -> None:
+        action_schema = self._convert_action_to_schema(action_space)
+        config = ChatGPTConfig(
+            tool_choice="required", tools=action_schema, **self.parameters
```
Comment (dandansamax) [inline crab/agents/backend_models/camel_model.py:87]: The `tool_choice` parameter depends on if the input `action_space` is None.

File: crab/agents/backend_models/camel_model.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        )
+        self.client = ExternalToolAgent(
+            model=backend_model,
+            system_message=sysmsg,
+            external_tools=action_schema,
+            message_window_size=self.history_messages_len,
+        )
+        self.token_usage = 0
+
+    # TODO: convert Action into OpenAIFunction
```
Comment (dandansamax) [inline crab/agents/backend_models/camel_model.py:106]: What does TODO mean here?

File: crab/agents/backend_models/camel_model.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        for message in messages:
+            if message[1] == MessageType.IMAGE_JPG_BASE64:
+                image = decode_image(message[0])
+                image_list.append(image)
+            else:
+                content = message[0]
+        usermsg = BaseMessage.make_user_message(
+            role_name="User",
+            content=content,
+            image_list=image_list,
```
Comment (dandansamax) [inline crab/agents/backend_models/camel_model.py:146]: Is it possible to make the images between two text content like OpenAI API does?

**Review by dandansamax — CHANGES_REQUESTED** [review]
Leave some small fix. 

Besides, a big possible improvement can be, instead of build the agent inside the `CamelModel`, can we pass the agent into the CamelModel so that we can construct the agent with any component outside of crab?

File: crab/agents/backend_models/camel_model.py (hunk 4 above)
Comment (WHALEEYE) [reply crab/agents/backend_models/camel_model.py:146]: Not available for now, but CAMEL is refactoring the message and this will be available afterwards

Comment (dandansamax) [reply crab/agents/backend_models/camel_model.py:146]: Sure. Let's wait

**Comment by dandansamax** [comment]
Please run `poetry lock` to fix the test

**Review by dandansamax — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**